src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -708,5 +710,155 @@ public static nuint RotateRight(nuint value, int offset)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+
+            Span<byte> bytes = stackalloc byte[sizeof(ushort)];
+            Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(bytes), data);
+
+            foreach (byte b in bytes)
+            {
+                int tableIndex = (int)((crc ^ b) & 0xFF);
+                crc = s_crcTable[tableIndex] ^ (crc >> 8);
+            }
+
+            return crc;
+        }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * unsigned int _mm_crc32_u32 (unsigned int crc, unsigned int v)
+        /// * uint32_t __crc32ch (uint32_t a, uint32_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, uint data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // unsigned int _mm_crc32_u32 (unsigned int crc, unsigned int v)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32ch (uint32_t a, uint32_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+
+            Span<byte> bytes = stackalloc byte[sizeof(uint)];
+            Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(bytes), data);
+
+            foreach (byte b in bytes)
+            {
+                int tableIndex = (int)((crc ^ b) & 0xFF);
+                crc = s_crcTable[tableIndex] ^ (crc >> 8);
+            }
+
+            return crc;
+        }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * unsigned __int64 _mm_crc32_u64 (unsigned __int64 crc, unsigned __int64 v)
+        /// * uint32_t __crc32d (uint32_t a, uint64_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, ulong data)
+        {
+            if (Sse42.X64.IsSupported)
+            {
+                // unsigned int _mm_crc32_u32 (unsigned int crc, unsigned int v)
+                ulong result = Sse42.X64.Crc32(crc, data);
+
+                return unchecked((uint)result);
+            }
+            if (Crc32.Arm64.IsSupported)
+            {
+                // uint32_t __crc32d (uint32_t a, uint64_t b)
+                return Crc32.Arm64.ComputeCrc32(crc, data);
```

Comment:
Shouldn't this use `Crc32.Arm64.ComputeCrc32C`?

Reply:
Yes. It should. Thank you, fixed in last commit.